src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/controller/ListingController.java

```diff
@@ -120,4 +129,165 @@ public void deleteListing(@PathVariable UUID id) {
+                    return new ResponseStatusException(HttpStatus.NOT_FOUND,
+                            "listing with id " + id + " not found");
+                });
+
+        ApplicationGetDto application =  applicationService.findApplicationById(applicationID)
+                .map(ApplicationMapper.INSTANCE::applicationToApplicationGetDto)
+                .orElseThrow(() ->
+                        new ResponseStatusException(HttpStatus.NOT_FOUND, "application with id " + applicationID + " not found"));
+
+        if (application.getListing().getId() != listing.getId()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "application to listing couldn't be found");
+        }
+        // either publisher of listing to see application or applicant itself
+        if ((authUser.getId() != application.getApplicant().getId()) &&
+                (authUser.getId() != listing.getPublisher().getId())) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "not allowed to get applications to listing");
+        }
+
+        return application;
+    }
+
+    // publisher of listing changes status of an application
+    @PutMapping("/listings/{id}/applications/{applicationID}")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public ApplicationGetDto updateApplication(
+            @AuthenticationPrincipal String authUserId,
+            @PathVariable UUID id,
+            @PathVariable UUID applicationID,
+            @RequestBody ApplicationPostPutDto applicationPostPutDto) {
+        User authUser = userService.findUserID(UUID.fromString(authUserId))
+                .orElseThrow(() -> {
+                    log.error("user (authenticated user) with id {} not found while getting application", authUserId);
+                    return new ResponseStatusException(HttpStatus.FORBIDDEN,
+                            "error finding authenticated user");
+                });
+
+        Listing listing = listingService.findListingById(id)
+                .orElseThrow(() -> {
+                    log.debug("listing with id {} not found while geting listing", id);
+                    return new ResponseStatusException(HttpStatus.NOT_FOUND,
+                            "listing with id " + id + " not found");
+                });
+
+        Application applicationToBeUpdated = applicationService.findApplicationById(applicationID)
+                .orElseThrow(() -> {
+                    log.error("application with id {} not found", applicationID);
+                    return new ResponseStatusException(HttpStatus.NOT_FOUND,
+                            "error finding application");
+                });
+
+        Application applicationInput = ApplicationMapper.INSTANCE.applicationPostPutDtoToApplication(applicationPostPutDto);
+
+        return ApplicationMapper.INSTANCE.applicationToApplicationGetDto(
+                applicationService.decideOnApplication(applicationInput, authUser));
```

Comment:
note: for better readability, I'd move the `applicationService.decideOnApplication(applicationInput, authUser)` out and create a variable. but this is of course fine, too